### PR TITLE
Generate CLI reference for documentation

### PR DESCRIPTION
**Pull request by zanieb**
Loosely based on [Cargo's format](https://github.com/rust-lang/cargo/blob/master/src/doc/src/commands/cargo-build.md)

Screenshot 2024-08-01 at 9 44 03 AM

Future work includes:

- Grouping options
- Enforcing some sort of specific command ordering
- Showing possible values for enums
- Adding "long_about" to commands for more context

### CODE REVIEW

**Comment by zanieb** [comment]
Notably the top-level preview commands are excluded because they're hidden

File: docs/stylesheets/extra.css [hunk 1]
```diff
@@ -147,3 +147,20 @@ See https://github.com/astral-sh/uv/issues/5130 */
 .md-nav__link--active {
   font-weight: bold;
 }
+
+/* Styling for the generated CLI reference page */
```
Comment (zanieb) [inline docs/stylesheets/extra.css:151]: I created a custom class, these have ~no reliance on mkdocs material

**Review by ibraheemdev — APPROVED** [review]
Love this.

**Comment by eth3lbert** [comment]
Great! I think I can help with future work part once we've finalized the grouping and ordering.

**Pull request merged**